intake_sql/sql_cat.py

```diff
@@ -32,7 +32,7 @@ def _load(self):
                 if c.primary_key:
                     description = 'SQL table %s from %s' % (name, self.uri)
                     args = {'uri': self.uri, 'table': name, 'index': c.name,
-                            'sql_kwargs': self.kwargs}
+                            'sql_kwargs': {}}
```

Comment:
Why this change, should we not be able to pass kwargs to the loader?

Reply:
I ran into troubles here testing with SQLite.  `self.kwargs` included both `"name"` and `"ttl"`, neither of which were valid kwargs for SQLAlchemy. It's not clear to me what the intent of these `kwargs` was, but that seemed overly broad to me.

Perhaps driver should take a `sql_kwargs` argument in its `args`, which will be passed to the table reader?

Reply:
Ah, I see that this was fixed in my local branch but never pushed. What do you think of this:
```diff
--- a/intake_sql/sql_cat.py
+++ b/intake_sql/sql_cat.py
@@ -15,7 +15,8 @@ class SQLCatalog(Catalog):
     name = 'sql_cat'
     version = __version__

-    def __init__(self, uri, views=False, **kwargs):
+    def __init__(self, uri, views=False, sql_kwargs=None, **kwargs):
+        self.sql_kwargs = sql_kwargs or {}
         self.uri = uri
         self.views = views
         super(SQLCatalog, self).__init__(**kwargs)
@@ -32,7 +33,7 @@ class SQLCatalog(Catalog):
                 if c.primary_key:
                     description = 'SQL table %s from %s' % (name, self.uri)
                     args = {'uri': self.uri, 'table': name, 'index': c.name,
-                            'sql_kwargs': self.kwargs}
+                            'sql_kwargs': self.sql_kwargs}
                     e = LocalCatalogEntry(name, description, 'sql_auto', True,
                                           args, {}, {}, {}, "", getenv=False,
                                           getshell=False)
```

We generally want to allow general kwargs which may be ignored or picked up by the base class, to allow for sources accessed via a server, for example.

Reply:
Yes, this diff looks right to me. If you push it I can rebase this PR against it.

Reply:
Added this diff, but I can deal with the conflict if you prefer to push your commit.